backend/swift/swift.go

```diff
@@ -632,6 +641,16 @@ func (f *Fs) listContainerRoot(ctx context.Context, container, directory, prefix
 				if addContainer {
 					remote = path.Join(container, remote)
 				}
+
+				if isDirectory {
+					// Swift returns duplicate directories when a directory marker exists, hence dedupe is required
+					remoteTrimmed := strings.TrimRight(remote, "/")
+					if _, found := dirs[remoteTrimmed]; found {
+						continue
+					}
+					dirs[remoteTrimmed] = true
+				}
```

Comment:
This can probably be avoided if the Swift library doesn't return duplicated objects. If I have a directory marker `empty-dir`, Swift will return the following payload:

```
[
  {
    "bytes": 0,
    "hash": "d41d8cd98f00b204e9800998ecf8427e",
    "name": "empty-dir",
    "content_type": "application/directory",
    "last_modified": "2023-06-14T11:49:19.990560"
  },
  {
    "subdir": "empty-dir/"
  }
]
```
Please find the request log [here](https://gist.github.com/joec4i/b79afd610d96457d6c62eb68f315b6ef). 

At the moment, the go Swift library will translate the response into two `swift.Object`s with the same name. So we need the deduplication step here to avoid outputting duplicated entries.

[The Swift client library](https://github.com/ncw/swift) is probably a better place to dedupe though.